src/boot/main.c

```diff
@@ -303,13 +303,19 @@ void handle_dp_complete(void) {
     sCurrentDisplaySPTask = NULL;
 }
 
-OSTimer RCPHangTimer;
+OSTimerEx RCPHangTimer;
+int osSetTimer_log(OSTimer* t, OSTime countdown, OSTime interval, OSMesgQueue* mq, OSMesg msg);
 void start_rcp_hang_timer(void) {
-    osSetTimer(&RCPHangTimer, OS_USEC_TO_CYCLES(3000000), (OSTime) 0, &gIntrMesgQueue, (OSMesg) MESG_RCP_HUNG);
+    if (RCPHangTimer.started == FALSE) {
+        osSetTimer(&RCPHangTimer.timer, OS_USEC_TO_CYCLES(3000000), (OSTime) 0, &gIntrMesgQueue, (OSMesg) MESG_RCP_HUNG);
+        RCPHangTimer.started = TRUE;
+    }
 }
 
+void osStopTimer_log(OSTimer *);
 void stop_rcp_hang_timer(void) {
-    osStopTimer(&RCPHangTimer);
+    osStopTimer(&RCPHangTimer.timer);
+    RCPHangTimer.started = FALSE;
```

Comment:
Does this also need a conditional check or does it not matter?

Reply:
Stopping a timer twice seems to work just fine, it's starting one twice that breaks things